src/daft-core/src/array/ops/json.rs

```diff
@@ -0,0 +1,89 @@
+use crate::datatypes::Utf8Array;
+use arrow2;
+
+use super::as_arrow::AsArrow;
+use common_error::{DaftError, DaftResult};
+use itertools::Itertools;
+use jaq_interpret::{Ctx, FilterT, ParseCtx, RcIter};
+use serde_json::Value;
+
+impl Utf8Array {
+    pub fn json_query(&self, query: &str) -> DaftResult<Utf8Array> {
+        // parse the query string
+        let (filter, errs) = jaq_parse::parse(query, jaq_parse::main());
+        if !errs.is_empty() {
+            return Err(DaftError::ValueError(format!(
+                "Error parsing json query ({query}): {}",
+                errs.iter().map(|e| e.to_string()).join(", ")
+            )));
+        }
+
+        // set up the parse context with the core and std libraries https://github.com/01mf02/jaq/tree/main?tab=readme-ov-file#features
```

Comment:
Wonder how expensive this is? If it's significant, maybe we should wrap this in lazy_static

Reply:
Not too sure if lazy_static is possible? Since the filter compilation takes in the query, which is determined at runtime. 
Also I couldn't find any documentation on how expensive the compilation is, but i did a little test with a slightly more complex query: 
```.people | map(select(.age >= 15) | {name, votingAge: (.age >= 18)}) | sort_by(.votingAge) | reverse``` 
on this data :
```{"people": [{"name": "Alice", "age": 25, "city": "New York"}]}```

and these are the timings:
parse time: 10.514792ms
compile time: 8.084375ms
run time: 5.723333ms

Reply:
Oh yes not the filters, was thinking more of lazy_static for the ParseCtx definition. 

For the filter itself we maybe we could compile it when the expression is constructed and have it in the expression as a field instead of dynamically constructing at execution time. 

Could that work? I think it might rely on the jaq stuff being serializable though, since expressions have to be serializable.

Reply:
yeah the jaq stuff isn't serializable :( but yeah I could do lazy_static for the ParseCtx

Reply:
We could technically perform our own serialization/deserialization:

1. On serialization, store the query as a string
2. On deserialization, restore the compiled jaq datastructures by compiling the string

This might be overoptimization though, since it's 8ms for the entire partition

Reply:
Agreed. Also if the amount of rows is very large the execution time will be much longer than the compilation time. 
Just tried out the query above on 10000 rows:
compile time: 5.025959ms
run time: 290.031875ms